packages/fast-components-react-base/src/auto-suggest/auto-suggest.tsx

```diff
@@ -201,11 +212,41 @@ class AutoSuggest extends Foundation<
                     listbox__disabled: autoSuggest__disabled,
                 }}
             >
-                {this.props.children}
+                {this.renderChildren()}
             </Listbox>
         );
     }
 
+    private renderChildren(): React.ReactNode {
+        if (this.props.filterSuggestions) {
+            const children: React.ReactNode = this.props.children;
+            return React.Children.map(
+                children,
+                (node: React.ReactElement<any>): React.ReactNode | null => {
+                    if (node.props !== undefined) {
+                        if (node.props[AutoSuggest.valuePropertyKey] === undefined) {
+                            return node;
+                        }
+                        return this.isMatch(node.props) ? node : null;
```

Comment:
This does imply children are directly nested in order to be filtered, @scomea can you weigh in on this.

Reply:
AutoSuggest uses ListBox and ListBoxItem, structurally you need to directly nest ListBoxItem inside Listbox

Reply:
The data input for autosuggest is "children", so any operation done in that component will consist of evaluating this.props.children somehow.  I think this is fine for "local typeahead" type implementations like a relatively small list of options like "enter your state".  When the component is used for larger datasets I would not expect filtering to happen in the component but rather upstream before children are created (a controlled scenario).